modules/mediacodec_dec/mediacodec_dec.c

```diff
@@ -0,0 +1,891 @@
+        ctx->esd= esd;
+        if (!esd->decoderConfig->decoderSpecificInfo || !esd->decoderConfig->decoderSpecificInfo->data) {
+            ctx->width=ctx->height=128;
+            ctx->out_size = ctx->width*ctx->height*3/2;
+            ctx->pix_fmt = GF_PIXEL_NV21;
+        } else {
+            return MCDec_InitDecoder(ctx);
+        }
+    }
+
+    return MCDec_InitDecoder(ctx);
+}
+
+
+
+
+static GF_Err MCDec_DetachStream(GF_BaseDecoder *ifcg, u16 ES_ID)
+{
+    MCDec *ctx = (MCDec *)ifcg->privateStack;
+    
+#ifdef GPAC_ANDROID
+
+    if(AMediaCodec_stop(ctx->codec) != AMEDIA_OK) {
+        LOGE("AMediaCodec_stop failed");
+    }
+
+#endif
+
+    return GF_OK;
+}
+
+
+static GF_Err MCDec_GetCapabilities(GF_BaseDecoder *ifcg, GF_CodecCapability *capability)
+{
+
+    MCDec *ctx = (MCDec *)ifcg->privateStack;
+    
+    switch (capability->CapCode) {
+    case GF_CODEC_RESILIENT:
+        capability->cap.valueInt = 1;                   
+        break;
+    case GF_CODEC_WIDTH:
+        capability->cap.valueInt = ctx->width;          
+        break;
+    case GF_CODEC_HEIGHT:
+        capability->cap.valueInt = ctx->height;         
+        break;
+    case GF_CODEC_STRIDE:
+        capability->cap.valueInt = ctx->stride;         
+        break;
+    case GF_CODEC_FPS:
+        capability->cap.valueFloat = ctx->frame_rate;
+        break;
+    case GF_CODEC_PAR:
+        capability->cap.valueInt = ctx->pixel_ar;
+        break;
+    case GF_CODEC_OUTPUT_SIZE:
+
+        capability->cap.valueInt = ctx->out_size;
+        break;
+    case GF_CODEC_PIXEL_FORMAT:
+        capability->cap.valueInt = ctx->pix_fmt;        
+        break;
+    case GF_CODEC_BUFFER_MIN:
+        capability->cap.valueInt = 1;                   
+        break;
+    case GF_CODEC_BUFFER_MAX:
+        capability->cap.valueInt = 6;
+        break;
+    /*by default we use 4 bytes padding (otherwise it happens that XviD crashes on some videos...)*/
+    case GF_CODEC_PADDING_BYTES:
+        capability->cap.valueInt = 0;                   
+        break;
+    /*reorder is up to us*/
+    case GF_CODEC_REORDER:
+        capability->cap.valueInt = 0;                   
+        break;
+    case GF_CODEC_WANTS_THREAD:
+        capability->cap.valueInt = 0;                   
+        break;
+    case GF_CODEC_FRAME_OUTPUT:
+        capability->cap.valueInt = 1;                   
+        break;
+    
+    /*not known at our level...*/
+    case GF_CODEC_CU_DURATION:                             
+    default:
+        capability->cap.valueInt = 0;
+        break;
+    }
+    return GF_OK;
+}
+
+
+static GF_Err MCDec_SetCapabilities(GF_BaseDecoder *ifcg, GF_CodecCapability capability)
+{
+    return GF_NOT_SUPPORTED;
+}
+
+
+void convertNV12toNV21(char *bytes, u32 width, u32 height) 
+{
+    u32 i;
+    for(i = width*height; i < 3 * (width * height)/ 2 - 1; i++){
+        bytes[i] = bytes[i + 1];
```

Comment:
There is no support for NV12 in GPAC yet. We could take every NV21 occurence and add NV12 ; there doesn't seem to be that much occurrences: https://github.com/gpac/gpac/search?utf8=%E2%9C%93&q=nv21
@rauf Is it clear for you?